File: Lession 2/SelectionSort.jl
function SelectionSort(arr)
    n = length(arr)
    for i in 1:n
        minIndex = i
        for j in i+1:n
            if arr[j] < arr[minIndex]
                minIndex = j
            end
        end
        arr[i], arr[minIndex] = arr[minIndex], arr[i]
    end
    return arr
end

arr = [64, 25, 12, 22, 11]
println(SelectionSort(arr))

File: Lession 2/SelectionSort.cpp
#include <iostream>
using namespace std;
void SelectionSort(int a[], int n)
{
    for (int i = 0; i < n - 1; i++)
    {
        int min = i;
        for (int j = i + 1; j < n; j++)
        {
            if (a[j] < a[min])
            {
                min = j;
            }
        }
        if (min != i)
        {
            int temp = a[i];
            a[i] = a[min];
            a[min] = temp;
        }
    }
}
int main()
{
    int a[] = {5, 2, 6, 1, 3, 4};
    int n = sizeof(a) / sizeof(a[0]);
    SelectionSort(a, n);
    for (int i = 0; i < n; i++)
    {
        cout << a[i] << " ";
    }
    return 0;
}